File: src/main/java/com/patneh/shelter/infrasructure/enums/StanEnum.java

package com.patneh.shelter.infrasructure.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public enum StanEnum {
    NOWY("Nowy"),
    DO_SPRAWDZENIA("Do_sprawdzenia"),
    DO_ADOPCJI("Do_adopcji"),
    ADOPTOWANY("Adoptowany");

    @Getter
    private final String label;
}

File: shelter-frontend/src/app/shared/enums/stan.enum.ts
export enum StanEnum {
  NOWY = 'NOWY',
  DO_SPRAWDZENIA = 'DO_SPRAWDZENIA',
  DO_ADOPCJI = 'DO_ADOPCJI',
  ADOPTOWANY = 'ADOPTOWANY',
}

export const StanLabels: Record<StanEnum, string> = {
  [StanEnum.NOWY]: 'Nowy',
  [StanEnum.DO_SPRAWDZENIA]: 'Do sprawdzenia',
  [StanEnum.DO_ADOPCJI]: 'Do adopcji',
  [StanEnum.ADOPTOWANY]: 'Adoptowany'
};
